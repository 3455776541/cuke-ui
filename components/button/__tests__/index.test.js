import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "../index";

describe("<Button/>", () => {
	it("should render a <Button/> components", () => {
		const wrapper = render(
			<div>
				<Button>你好</Button>
				<Button type="primary">你好</Button>
				<Button type="info">你好</Button>
				<Button type="warning">你好</Button>
				<Button type="error">你好</Button>
				<Button type="success">你好</Button>
				<Button disabled>你好</Button>
				<Button loading>加载中...</Button>
				<Button block>100%</Button>
				<Button type="primary" size="large">
					大
				</Button>

				<Button type="info">中</Button>

				<Button type="warning" size="small">
					小
				</Button>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-button classnames", () => {
		const wrapper = render(
			<div>
				<Button>你好</Button>
				<Button type="primary">你好</Button>
				<Button type="info">你好</Button>
				<Button type="warning">你好</Button>
				<Button type="error">你好</Button>
				<Button type="success">你好</Button>
				<Button disabled>你好</Button>
				<Button loading>加载中...</Button>
				<Button block>100%</Button>
			</div>
		);
		assert(wrapper.find(".cuke-button").length >= 1);
		assert(wrapper.find(".btn-primary").length === 1);
		assert(wrapper.find(".btn-info").length === 1);
		assert(wrapper.find(".btn-warning").length === 1);
		assert(wrapper.find(".btn-error").length === 1);
		assert(wrapper.find(".btn-success").length === 1);
		assert(wrapper.find(".btn-disabled").length === 1);
		assert(wrapper.find(".btn-loading").length === 1);
		assert(wrapper.find(".btn-block").length === 1);
	});

	it("should can trigger click event", () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Button onClick={onClick}>黄瓜ui</Button>);
		wrapper.find("button").simulate("click");
		expect(onClick).toHaveBeenCalled();
	});
	it("should can not trigger click event", () => {
		const onClick = jest.fn();
		const wrapper = shallow(
			<Button onClick={onClick} disabled>
				黄瓜ui
			</Button>
		);
		wrapper.find("button").simulate("click");
		expect(onClick).not.toHaveBeenCalled();
	});

	it("should render link", () => {
		const wrapper = shallow(<Button href="/test">黄瓜ui</Button>);
		assert(wrapper.find("a.cuke-button").length >= 1);
	});
});
