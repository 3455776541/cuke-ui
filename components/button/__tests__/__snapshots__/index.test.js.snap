// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render a <Button/> components 1`] = `
<div>
  <button
    class="cuke-button btn-default"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-primary"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-info"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-warning"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-error"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-success"
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-default btn-disabled"
    disabled=""
    type="button"
  >
    <span>
      你好
    </span>
  </button>
  <button
    class="cuke-button btn-default btn-loading"
    disabled=""
    type="button"
  >
    <svg
      attr="[object Object]"
      class="undefined cuke-loading"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <line
        x1="12"
        x2="12"
        y1="2"
        y2="6"
      />
      <line
        x1="12"
        x2="12"
        y1="18"
        y2="22"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="4.93"
        y2="7.76"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="16.24"
        y2="19.07"
      />
      <line
        x1="2"
        x2="6"
        y1="12"
        y2="12"
      />
      <line
        x1="18"
        x2="22"
        y1="12"
        y2="12"
      />
      <line
        x1="4.93"
        x2="7.76"
        y1="19.07"
        y2="16.24"
      />
      <line
        x1="16.24"
        x2="19.07"
        y1="7.76"
        y2="4.93"
      />
    </svg>
    <span>
      加载中...
    </span>
  </button>
  <button
    class="cuke-button btn-default btn-block"
    type="button"
  >
    <span>
      100%
    </span>
  </button>
</div>
`;
